src/main/java/seedu/address/logic/commands/UpcomingCommand.java

```diff
@@ -1,32 +1,39 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_DATE;
+
+import java.util.List;
+import java.util.function.Predicate;
 
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
-import seedu.address.model.delivery.DeliveryIsUpcomingPredicate;
-
+import seedu.address.model.delivery.Delivery;
 /**
  * Lists all pending deliveries that are completed before a specified date.
  */
 public class UpcomingCommand extends Command {
     public static final String COMMAND_WORD = "upcoming";
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + " : Lists all pending deliveries with completion date before the specified date.\n"
-            + "Parameters: Completion_Date_Time (dd-mm-yyyy hh:mmr)\n"
-            + "Example: " + COMMAND_WORD + " 18-06-2023 17:00";
+            + " : Lists all pending deliveries with completion date before and after the specified dates.\n"
+            + "Parameters: "
+            + PREFIX_START_DATE + "START (optional) "
```

Comment:
Perhaps it is better to state separately that for any command, only one is optional?